packages/graph/hash_graph/lib/graph/src/store/postgres/resolve.rs

```diff
@@ -0,0 +1,161 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use tokio_postgres::{GenericClient, Row, RowStream};
+use type_system::{uri::VersionedUri, DataType, PropertyType};
+
+use crate::store::{postgres::parameter_list, AsClient, PostgresStore, QueryError};
+
+#[async_trait]
+pub trait PostgresContext {
```

Comment:
Would be nice to also have an explanation for this trait

Reply:
I added a short explanation in f52d40f07e4b8df4cad07cc51d1ee24d1a53aaaf. As the trait is only used internally for implementing `Resolve` on `Record<DataType>` and `Record<PropertyType>` and I expect to change it as we proceed, we should probably not spend too much time on it. The blanket implementation for `&T` below is only for simplifying its usage for `&PostgresStore`, in the future I expect a custom struct for this.